react/features/conference/middleware.js

```diff
@@ -35,6 +39,12 @@ MiddlewareRegistry.register(store => next => action => {
 
         break;
     }
+    case KICKED_OUT: {
+        store.dispatch(
+            conferenceFailed(action.conference, JitsiConferenceEvents.KICKED));
```

Comment:
@paweldomas, the only reason why I'm delaying on this one is because I'm still battling with myself whether it should be `conferenceFailed` or `conferenceLeft` (as you and I have been pondering from the onset of this task).

Reply:
Sorry to jump in, but I don't know what are the concerns about `conferenceFailed` or `conferenceLeft`, but just to mention that in xmpp this is not an error. The user receives an unavailable presence with reason and a status code to indicate the kick. So looking it from xmpp perspective this is just `conferenceLeft `.
https://xmpp.org/extensions/xep-0045.html#kick

Reply:
@lyubomir FWIW, here are my 2 cents, based on my comment here: https://github.com/jitsi/jitsi-meet/pull/2632#discussion_r178092328 I think it should be failed. There are other cases in which the conference might fail but we don't retry with the overlay, so navigating back and showing some error (imagine a toast) sounds like a good idea to me. Not as part of this PR, but this PR could take a small step in that direction.